RedHatInsights/patchman-ui
# SPM-1729: Show only relevant OS filter options across tables

Implements https://issues.redhat.com/browse/SPM-1729.

This changes the behavior of the OS filter across tables. The filter now only shows the options that are relevant and available for the user's account. The options are requested from the Inventory API.

## How to test

1. Check the following pages and components to contain the same values for the OS filter:
1.1. https://stage.foo.redhat.com:1337/beta/insights/patch/systems
1.2. https://stage.foo.redhat.com:1337/beta/insights/patch/advisories/${advisory_id}
1.3. https://stage.foo.redhat.com:1337/beta/insights/patch/templates: create patch template wizard component.
2. Check whether a call to the Inventory API's /operating_system is made in all pages.
3. Check chips/filter selection behavior and URL parameters update.

## Screenshots

![Screenshot 2022-09-15 at 15-27-15 Patch template - Patch Red Hat Insights](https://user-images.githubusercontent.com/31385370/190415978-caffe7d1-73f1-492e-94a9-82a16026195f.png)
![Screenshot 2022-09-15 at 15-26-53 FEDORA-EPEL-2022-929440e4dc - Advisories - Patch Red Hat Insights](https://user-images.githubusercontent.com/31385370/190415985-1bdea00b-26d1-42cb-96d1-ae55b247f237.png)
![Screenshot 2022-09-15 at 15-26-19 Systems - Patch Red Hat Insights](https://user-images.githubusercontent.com/31385370/190415990-1f4243de-0510-409f-82b4-d561073ff422.png)
